### PR TITLE
Fix reservations handling in logs 

### LINKED ISSUE #10189: Sonic Unleashed Crash Regression
## Quick summary
Sonic Unleashed crashes are more common and vary greatly in the latest build of RPCS3 (0.0.15-12155-7197dd75 Alpha).

## Details
In build 0.0.11-10736-d3585e1f Alpha, Sonic Unleashed works on my hardware using default settings, with low performance.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/6371139/RPCS3.log.gz)

In build 0.0.11-10736-d3585e1f Alpha, Sonic Unleashed works on my hardware using the recommended RPCS3 Wiki Settings (1us Driver Wake-Up Delay), with slightly better performance.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/6371141/RPCS3.log.gz)

In build 0.0.15-12155-7197dd75 Alpha, Sonic Unleashed crashes randomly on my hardware using the recommended RPCS3 Wiki Settings (Tried changing **Driver Wake-Up Delay** between _1us_ and _200us_, tried changing **Shader Mode** between _Async (multi threaded)_ and _Async with Shader Interpreter_).
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/6371170/RPCS3.log.gz)

#### 1. Please provide _exact_ build (or commit) information that introduced the regression you're reporting.
0.0.15-12155-7197dd75 Alpha

#### 2. Please attach TWO logs:
* One for the build with regression.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/6371170/RPCS3.log.gz)
* One for the build that works as expected.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/6371141/RPCS3.log.gz)

#### 3. If you describe graphical regression, please provide an RSX capture and a RenderDoc capture that demonstrate it.
Not a graphical regression.

#### 4. Please attach screenshots of your problem.
Screenshots are not effective to describe the issue, the game will freeze-frame and stay there, while an error is logged on RPCS3.

#### 5. Please provide comparison with real PS3.
PS3 doesn't crash randomly.

#### 6. Please provide your system configuration:
* Windows 10 (64-bit)
* Intel i7-6700K (4Ghz)
* AMD RX 580
* Radeon Software Version 20.11.1

#### Please include.
Prior to attempting to write these logs, the game would crash on areas like the second cutscene, first cutscene, loading screens, etc. Upon generating the log to upload here for the latest RPCS3, it went way further than normal with no game configuration changes (I got all the way up to the end of the second stage before it crashed).

These crashes are not the purple crash (Not logged as a fatal error), it's a "red crash" (Logged as a regular error).

### PR DESCRIPTION
**Pull request by elad335**
Fix CELL reservations support by logs.
Fixes #10189 and maybe more stuff.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellPad.cpp [hunk 1]
```diff
@@ -456,40 +475,6 @@ error_code cellPadPeriphGetInfo(vm::ptr<CellPadPeriphInfo> info)
...
-
-	auto& config = g_fxo->get<pad_info>();
-
-	if (!config.max_connect)
-		return CELL_PAD_ERROR_UNINITIALIZED;
-
-	const auto handler = pad::get_current_handler();
-
-	// port_no can only be 0-6 in this function
-	if (port_no >= CELL_PAD_MAX_PORT_NUM || !data)
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellPad.cpp:473]: What happened to this check?

File: rpcs3/Emu/Cell/Modules/cellPad.cpp [hunk 2]
```diff
@@ -410,6 +420,15 @@ error_code cellPadGetData(u32 port_no, vm::ptr<CellPadData> data)
 		}
 	}
 
+	if (const auto ppu = get_current_cpu_thread<ppu_thread>(); ppu &&
+		ppu->cia == g_fxo->get<ppu_function_manager>().func_addr(FIND_FUNC(cellPadPeriphGetData)) + 4)
+	{
+		// For cellPadPeriphGetData
+		const auto extra = vm::_ptr<CellPadPeriphData>(static_cast<u32>(ppu->gpr[4]));
+		extra->pclass_type = pad->m_class_type;
```
[the remark this answers is not in the document]
Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellPad.cpp:428]: I disagree with you, the functionality had to moved to "here",  the usage of recusrive mutex was horrible.

File: rpcs3/Emu/Cell/Modules/cellPad.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellPad.cpp:473]: Retrieved.

File: rpcs3/Emu/Cell/Modules/cellPad.cpp [hunk 3]
```diff
@@ -137,6 +137,27 @@ error_code cellPadClearBuf(u32 port_no)
...
+
+	// Don't lock mutex here!
+
+	auto& config = g_fxo->get<pad_info>();
+
+	if (!config.max_connect)
+		return CELL_PAD_ERROR_UNINITIALIZED;
+
+	// port_no can only be 0-6 in this function
+	if (port_no >= CELL_MAX_PADS)
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellPad.cpp:154]: Wrong constant

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellPad.cpp]: Fixed.

**Pull request merged**